src/intersection.js

```diff
@@ -23,5 +23,7 @@ var uniq = require('./uniq');
  *      R.intersection([1,2,3,4], [7,6,5,4,3]); //=> [4, 3]
  */
 module.exports = _curry2(function intersection(list1, list2) {
-  return uniq(_filter(flip(_contains)(list1), list2));
+  var lookupList = list1.length > list2.length ? list1 : list2;
+  var filteredList = lookupList == list1 ? list2 : list1;
```

Comment:
@CrossEye it's all a matter of integrating with babel ;)

@davidchambers so,
```js
if (...) {
  return ...uniq...filter..
} else {
  return ...uniq...filter..
}
```
or
```js
var lookupList, filteredList;
if (...) {
 // set them
}
else { ... }
return ...uniq...filter;
}
```
Because I honestly don't like either options, and would also not like to depend on ``var`` hoisting.